### PR TITLE
Improve pwa.js testing

### PR DESCRIPTION
**Pull request by juliantoledo**
Added more tests for lib/pwa.js (for #63)

### CODE REVIEW

File: test/lib/pwa.js [hunk 1]
```diff
@@ -32,39 +36,114 @@ var assert = require('chai').assert;
...
+      });
+    });
+  });
 
-          // Mock libImages and bd to avoid making real calls
-          simpleMock.mock(libImages, 'fetchAndSave').resolveWith(manifest.getBestIconUrl());
-          pwaLib.libImages = libImages;
-          simpleMock.mock(db, 'update').returnWith(pwa);
-          pwaLib.db = db;
+  describe('#save (core logic)', () => {
```
Comment (ithinkihaveacat) [inline test/lib/pwa.js]: 👍 

This is all so laborious—I couldn't be bothered doing it—glad we have something now!

File: test/lib/pwa.js [hunk 2]
```diff
@@ -32,39 +36,114 @@ var assert = require('chai').assert;
...
+        assert.equal(libPwa.fetchMetadataDescription.lastCall.args[0], pwa);
+        assert.equal(updatedPwa.metaDescription, 'test-description');
+        assert.equal(db.update.callCount, 1);
+        assert.equal(libPwa.updateIcon.callCount, 1);
+        assert.equal(libPwa.updateLighthouseInfo.callCount, 1);
+      });
+    });
+    it('handles E_MANIFEST_ERROR error', () => {
+      // Mock findByManifestUrl to avoid making real calls
+      simpleMock.mock(libPwa, 'findByManifestUrl').rejectWith(new Error('Testing error'));
```
Comment (ithinkihaveacat) [inline test/lib/pwa.js]: This results in an ugly stack trace in the middle of the mocha output, is there any way to get rid of this?

```text
Updated PWA Lighthouse info for:  123456789
      ✓ sets lighthouseScore
    #save (core logic)
      ✓ performs all the save steps
Error: Testing error
    at Context.it (/Users/stillers/workspace/gulliver/test/lib/pwa.js:133:63)
    at callFn (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runnable.js:334:21)
    at Test.Runnable.run (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runnable.js:327:7)
    at Runner.runTest (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:429:10)
    at /Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:535:12
    at next (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:349:14)
    at /Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:359:7
    at next (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:285:14)
    at Immediate.<anonymous> (/Users/stillers/workspace/gulliver/node_modules/mocha/lib/runner.js:327:5)
    at runCallback (timers.js:574:20)
    at tryOnImmediate (timers.js:554:5)
    at processImmediate [as _immediateCallback] (timers.js:533:5)
      ✓ handles E_MANIFEST_ERROR error
    #save (validation logic)
      ✓ rejects on null pwa
      ✓ rejects if not passed a Pwa object
      ✓ rejects if passed a Pwa object without a manifestUrl
      ✓ rejects if passed a Pwa object with an invalid manifestUrl
      ✓ rejects if passed a Pwa object with an invalid user.id
      ✓ fulfills if passed a valid Pwa objectid
```

Comment (juliantoledo) [inline test/lib/pwa.js]: @ithinkihaveacat let me move it to another place. I can also add the error codes here.

**Comment by juliantoledo** [comment]
@ithinkihaveacat fixed comments.

**Pull request merged**